stdlib/io-console/0/io-console.rbs

```diff
@@ -0,0 +1,137 @@
+class IO
+  class ConsoleMode
+    def echo=: ((true | false)) -> (true | false)
+    def raw: (?min: int, ?time: int, ?intr: (true | false)) -> self
+    def raw!: (?min: int, ?time: int, ?intr: (true | false)) -> self
+  end
+
+  # Returns an File instance opened console.
+  #
+  # If `sym` is given, it will be sent to the opened console with `args` and the
+  # result will be returned instead of the console IO itself.
+  def self.console: () -> File?
+                  | (:close) -> nil
+                  | (Symbol sym, *untyped args) -> top
```

Comment:
```suggestion
                  | (Symbol sym, *untyped args) -> untyped
```

We use `untyped` instead of `top` for types not determine.

Reply:
I want to show that all types are possible. It's not that I haven't decided.

This is how I interpreted it.
untyped: No type determined yet.
top: any type is possible.

https://github.com/ruby/rbs/blob/edfa19dc7517b0d87d44a144ffcdbc40e4883fde/docs/syntax.md#void-boolish-or-top
> `top` is anything else.

`IO.console(sym, *args)` can call any method like `Object#send`.

```rb
IO.console(:fileno)
#=> 9

IO.console(:path)
#=> "/dev/tty"
```

Should I still use `untyped` in this case?

Reply:
Thanks for your explanation. I understand what you mean. 

I think it should be `top` In principle. If it is `top`, the user needs to cast the returned value explicitly. It is a bit verbose but safer than `untyped`.

```ruby
# path is `top`
path = IO.console(:path)
"foo" + path # type error
```


However `Object#send`'s returned value is `untyped` in RBS. So I'm not sure which is appropriate for this case.

https://github.com/ruby/rbs/blob/f54d88e30f59da75f796a6fdf13e4cd99e9da8a5/core/object.rbs#L649


I have a meeting with @soutaro tomorrow, so I'll ask him about it. Please wait a while :pray:

Reply:
Hi @ksss,

I think it's better to go with `untyped` here. It's almost equivalent with `send`, and thus it should make sense to assume the Ruby developers want the same here.

I agree that there are a discussion about which is better `untyped` or `top` for cases like this. I'm currently assuming that forcing `case-when` every `send` method calls (or anything we have `untyped` today) looks overkill for most of the Ruby developers.

Reply:
Thank you @pocke and @soutaro .
I will think about it that way from now on.